pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py

```diff
@@ -65,13 +65,23 @@ def get_waveform(self):
         self.write("C{ch}:WF? ALL")
         response = self.read_bytes(count=-1, break_on_termchar=True)
         descriptor_dictionary = self.get_descriptor()
-        rawWaveform = list(
-            struct.unpack_from(
-                "%db" % descriptor_dictionary["numDataPoints"],
-                response,
-                offset=descriptor_dictionary["descriptorOffset"],
-            ),
-        )
+        if descriptor_dictionary["numDataPoints"] == 0:
+            rawWaveform = list(
+                struct.unpack_from(
+                    "%db" % descriptor_dictionary["pointsScreen"],
+                    response,
+                    offset=descriptor_dictionary["descriptorOffset"],
+                ),
+            )
+        else:
+            rawWaveform = list(
+                struct.unpack_from(
+                    "%db" % descriptor_dictionary["numDataPoints"],
```

Comment:
another idea to not repeat most of the code:
~~~python
data_points = descriptor_dictionary["numDataPoints"] if descriptor_dictionary["numDataPoints"] els escriptor_dictionary["pointsScreen"]
raw_waveform = list(... f"{data_points:d}b", response,...)
~~~

Reply:
Thanks for the suggestion. I implemented it in the code.